core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1317,6 +1317,16 @@ package object config {
           s" be less than or equal to ${ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH}.")
       .createWithDefault(64 * 1024 * 1024)
 
+  private[spark] val CHECKPOINT_DIR =
+    ConfigBuilder("spark.checkpoint.dir")
+      .doc(
+        "Equivalent with SparkContext.setCheckpointDir. If set, the path becomes" +
+          "the default directory for checkpointing. It can be overwritten by" +
+          "SparkContext.setCheckpointDir.")
```

Comment:
nit:
 ```suggestion
        "Set the default directory for checkpointing. It can be overwritten by " +
          "SparkContext.setCheckpointDir.")
```

(I missed this in my earlier pass, my bad)